Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orient_polygon_soup_extension.h

```diff
@@ -77,26 +76,137 @@ duplicate_non_manifold_edges_in_polygon_soup(PointRange& points,
   return inital_nb_pts==points.size();
 }
 
+/*!
+ * \ingroup PMP_orientation_grp
+ *
+ * orients each triangle of a triangle soup using the orientation of its
+ * closest non degenerate triangle in a triangle soup.
+ *
+ * \tparam Concurrency_tag enables sequential versus parallel orientation.
+ *                         Possible values are `Sequential_tag` (the default),
+ *                         `Parallel_if_available_tag`, and `Parallel_tag`.
+ * \tparam ReferencePointRange a model of the concepts `RandomAccessContainer` whose value type is the point type.
+ * \tparam ReferenceTriangleRange a model of the concept `RandomAccessContainer`
+ *                                whose `value_type` is a model of the concept `RandomAccessContainer`
+ *                                whose `value_type` is `std::size_t`and of size 3.
+ * \tparam PointRange a model of the concepts `RandomAccessContainer`
+ *                    and `BackInsertionSequence` whose value type is the point type.
+ * \tparam TriangleRange a model of the concept `RandomAccessContainer`
+ *                       whose `value_type` is a model of the concept `RandomAccessContainer`
+ *                       whose `value_type` is `std::size_t`and of size 3.
+ *
+ * \param ref_points the points of the reference soup
+ * \param ref_faces the triangles of the reference soup
+ * \param points the points of the soup
+ * \param faces the triangles of the soup
+ * \param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *   \cgalParamNBegin{point_map}
+ *     \cgalParamDescription{a property map associating points to the elements of the point set `points`}
+ *     \cgalParamType{a model of `ReadablePropertyMap` whose key type is the value type
+ *                    of the iterator of `PointRange` and whose value type is `geom_traits::Point_3`}
+ *     \cgalParamDefault{`CGAL::Identity_property_map<geom_traits::Point_3>`}
+ *   \cgalParamNEnd
+ *
+ *   \cgalParamNBegin{geom_traits}
+ *     \cgalParamDescription{an instance of a geometric traits class}
+ *     \cgalParamType{The traits class must provide the nested functor `Collinear_3` to check whether three points are collinear. }
+ *     \cgalParamDefault{a \cgal Kernel deduced from the point type, using `CGAL::Kernel_traits`}
+ *     \cgalParamExtra{The geometric traits class must be compatible with the vertex point type.}
+ *   \cgalParamNEnd
+ * \cgalNamedParamsEnd
+ *
+ * \attention The types of points in `PointRange`, `geom_traits`, and `vertex_point_map` must be the same.
+ */
+template <class Concurrency_tag = CGAL::Sequential_tag,
+          class ReferencePointRange, class ReferenceFaceRange, class PointRange, class FaceRange,
+          class NamedParameters = parameters::Default_named_parameters>
```

Comment:
Only one named parameter pack?

Reply:
Actually, I'd put none. Indeed, we can define a wrapper class that would turn any range + point map into a `PointRange`. Then you'll tell me, then why not doing it for `FaceGraph` too?

Reply:
I agree there should be two, but `orient_triangle_soup_with_reference_triangle_mesh()` only has one. Bad excuse, but I did not need the second so I aligned on the other function.